plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/PublishIssuesStep.java

```diff
@@ -855,7 +857,8 @@ protected ResultAction run() throws IOException, InterruptedException, IllegalSt
             ResultAction action = publisher.attachAction(step.getTrendChartType());
 
             if (!step.isSkipPublishingChecks()) {
-                WarningChecksPublisher checksPublisher = new WarningChecksPublisher(action, getTaskListener());
+                String checksName = Optional.ofNullable(getContext().get(ChecksInfo.class)).map(ChecksInfo::getName).orElse(null);
+                WarningChecksPublisher checksPublisher = new WarningChecksPublisher(action, getTaskListener(), checksName);
```

Comment:
I would rather prefer it look like:
```
WarningChecksPublisher checksPublisher = new WarningChecksPublisher(action, getTaskListener(), getContext().get(ChecksInfo.class));
```

Then we can place the logic inside of `WarningChecksPublisher` and reuse it for the `recordIssues` step. (`PublishIssuesStep` is a different step with the name `publishIssues`).